webapp/channels/src/reducers/views/scheduled_posts.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {AnyAction} from 'redux';
+import {combineReducers} from 'redux';
+
+import type {ScheduledPostsState} from '@mattermost/types/schedule_post';
+
+import {ActionTypes} from 'utils/constants';
+
+function byTeamId(state: ScheduledPostsState['byTeamId'] = {}, action: AnyAction) {
+    switch (action.type) {
+    case ActionTypes.SCHEDULED_POSTS_RECEIVED: {
+        const {scheduledPosts, teamId} = action.data;
+        return {
+            ...state,
+            [teamId]: scheduledPosts,
+        };
+    }
```

Comment:
Having this in the redux store can create interesting artifacts, specially on slow connections. For example...

I go to team A, load everything, including a scheduled post for a DM.
Go to team B, load everything, including a scheduled post for a DM.
On team B, remove the scheduled post for the DM.
Navigate to scheduled posts on team A
While team A is loading the new scheduled posts, the scheduled post for that DM shows until the new data is retrieved.

One solution would be to not have this in the redux state, but the component state. That still has some problems, since we should need this information somewhere to show the amount of scheduled messages on the LHS. 0/5 on how to deal with this.

Reply:
I see the problem is that I'm incorrectly saving all received scheduled posts under the teamId, and not separating the ones belonging to DMs and GMs. Let me see how I can fix it.

Reply:
Okay, so this is now updated. Theres is not a flag on the API to fetch DM/GM posts, and the response is categorised by team ID. SO this should be good now.